webui/react/src/components/kit/UserAvatar.tsx

```diff
@@ -2,11 +2,9 @@ import React from 'react';
 
 import Avatar, { Props as AvatarProps } from 'shared/components/Avatar';
 import useUI from 'shared/contexts/stores/UI';
-import { DetailedUser } from 'types';
-import { getDisplayName } from 'utils/user';
-
+import { getDisplayName, UserNameFields } from 'utils/user';
 export interface Props extends Omit<AvatarProps, 'darkLight' | 'displayName'> {
-  user?: DetailedUser;
+  user?: UserNameFields;
```

Comment:
question: Why was this changed? It seems like it causes ripple effects requiring typecasts elsewhere.

Reply:
Thanks for pointing this out. The typecasting I added was actually unnecessary, so I cleaned that up. Apart from that mistake, the `User` type has the most precise subset of fields we need here, so switching to that should minimize the amount of typecasting needed.